### ISSUE TITLE
Don't store default values in scenes and resources

### ISSUE THREAD

**Issue by jitspoe** [godotengine/godot-proposals#11297]
### Describe the project you are working on

Retro FPS

### Describe the problem or limitation you are having in your project

Often times, when updating the engine, there will be a massive list of files changed because some new default parameter was added, and all the tscn/tres/etc. files updated to add the new default values.

For example, a new debug_color was added for collision shapes, and now all my scenes have a bunch of changes that make it difficult to determine what *actual* changes there were vs. just new default values:

![Image](https://github.com/user-attachments/assets/7f3ee43d-8aee-4601-9c3d-012ba948cd70)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Unless the value is different from the default, it shouldn't be stored.  Another drawback to storing the default, is if you change the default, all existing resources with the default already stored in them will not have the new default.  So, for example, if the default color for collision debug was changed halfway through a project, then the half of the collisions would have different colors.

Storing the default values seems to just add unnecessary stuff to parse when reading resources which can impact load times as well.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In short, flag if things have changed from the default, and only write the things that are non-default.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Core engine behavior -- can't be worked around.

### Is there a reason why this should be core and not an add-on in the asset library?

Core engine behavior -- can't be worked around.

**Comment by dalexeev** [comment]
* It's a bug, see https://github.com/godotengine/godot/issues/100088.
* See also https://github.com/godotengine/godot/issues/80894 (only custom properties in resources).
* Thanks for the contribution nonetheless!

**Comment by jitspoe** [comment]
Ah, so it is a bug.  Seems there are a lot of things that save the default values.  Are there issues open for default values on shader materials and default values for .import files as well?

**Comment by Calinou** [comment]
>  Seems there are a lot of things that save the default values. Are there issues open for default values on shader materials and default values for .import files as well?

No, but storing default values for `.import` files and export options is usually desired as it allows us to change the defaults later on without breaking existing projects.
